web/Web/web/static/js/document-viewer.js

```diff
@@ -1244,9 +1259,14 @@ docView.prototype = {
     return this.triggerEvent("beforeDocumentLoad", [docid]);
   },
 
-  afterDocumentLoad: function(docid) {
+  afterDocumentLoad: function(docid, rel) {
```

Comment:
Let's just call `previouslyJudgedDocs` from here and get the rel value from inside the function. It's better to have just the docid parameter for this function.

Reply:
I thought accessing another .js file variable like `previouslyJudgedDocs` would be more problematic and opposes the original purpose of event trigger mechanism you mentioned. I'll arrange to change them back but I need to understand the architecture more thoroughly I guess.
I faced so many formatting issues with PyCharm defaults. The faulty code indents makes the code less readable I'm afraid so someday we might need a refactoring.